sdk/monitor/monitor-ingestion/review/monitor-ingestion.api.md

```diff
@@ -0,0 +1,54 @@
+## API Report File for "@azure/monitor-ingestion"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CommonClientOptions } from '@azure/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public (undocumented)
+export class FailedLogsIngestionError extends Error {
+    constructor(message: string, errors: Array<UploadLogsError>);
+    // (undocumented)
+    errors: Array<UploadLogsError>;
+}
+
+// @public
+export class LogsIngestionClient {
+    constructor(endpoint: string, tokenCredential: TokenCredential, options?: LogsIngestionClientOptions);
+    // @internal (undocumented)
+    splitDataToChunks(logs: Record<string, any>[]): any[];
+    upload(ruleId: string, streamName: string, logs: Record<string, any>[], options?: UploadOptions): Promise<UploadResult>;
+}
+
+// @public
+export interface LogsIngestionClientOptions extends CommonClientOptions {
+    apiVersion?: string;
+}
+
+// @public (undocumented)
+export interface UploadLogsError {
+    failedLogs: Record<string, unknown>[];
+    responseError: Error;
+}
+
+// @public
+export interface UploadOptions {
+    maxConcurrency?: number;
+}
+
+// @public
+export interface UploadResult {
+    errors: Array<UploadLogsError>;
+    uploadStatus: UploadStatus;
+}
+
+// @public
+export type UploadStatus =
```

Comment:
Can we build total failure into this instead of relying on `FailedLogsIngestionError` ?

Reply:
done